pmd-core/src/main/java/net/sourceforge/pmd/RuleContext.java

```diff
@@ -31,6 +30,7 @@
 
     private Report report = new Report();
     private File sourceCodeFile;
+    @Deprecated
```

Comment:
the field is private, we can simply remove it

Reply:
Indeed, I'll remove it.